clusterloader2/pkg/measurement/common/network/network_performance_measurement.go

```diff
@@ -0,0 +1,449 @@
+		if err := npm.framework.ApplyTemplatedManifests(networkTestRequestFilePath, npm.templateMapping); err != nil {
+			klog.Error(err)
+		}
+	}
+}
+
+func (npm *networkPerformanceMeasurement) populateWorkerPodList(data *workerPodData) {
+	if _, ok := npm.podInfo.workerPodMap[data.workerNode]; !ok {
+		npm.podInfo.workerPodMap[data.workerNode] = []workerPodData{}
+	}
+	npm.podInfo.workerPodMap[data.workerNode] = append(npm.podInfo.workerPodMap[data.workerNode], *data)
+}
+
+func (npm *networkPerformanceMeasurement) calculateMetricDataValue(dataElem *measurementutil.DataItem, metricIndex int) {
+	var aggregatePodPairMetrics []float64
+	var metricResponse []float64
+
+	npm.metricLock.Lock()
+	defer npm.metricLock.Unlock()
+
+	switch npm.podRatioType {
+	case oneToOne:
+		for _, metricResponse = range npm.metricVal {
+			if len(metricResponse) > 0 {
+				dataElem.Data[value] = metricResponse[metricIndex]
+			} else {
+				dataElem.Data[value] = 0
+			}
+		}
+		klog.Info("Metric value: ", dataElem.Data[value])
+	case manyToMany:
+		for _, metricResponse = range npm.metricVal {
+			if len(metricResponse) > 0 {
+				aggregatePodPairMetrics = append(aggregatePodPairMetrics, metricResponse[metricIndex])
+			}
+		}
+		percentile := getPercentile(aggregatePodPairMetrics)
+		dataElem.Data[perc05] = percentile.percentile05
+		dataElem.Data[perc50] = percentile.percentile50
+		dataElem.Data[perc95] = percentile.percentile95
+		klog.Info("Aggregate Metric value: ", aggregatePodPairMetrics)
+	}
+	klog.Infof("PodPairNumber: %v, Response received number: %v ", len(npm.podInfo.uniquePodPairList), len(npm.metricVal))
+}
+
+func (npm *networkPerformanceMeasurement) createResultSummary() {
+	var resultSummary testResultSummary
+	switch npm.protocol {
+	case protocolTCP:
+		npm.getMetricData(&resultSummary, tcpBandwidth, throughput)
+		resultSummary.protocol = protocolTCP
+	case protocolUDP:
+		npm.getMetricData(&resultSummary, udpPacketPerSecond, packetPerSecond)
+		npm.getMetricData(&resultSummary, udpJitter, jitter)
+		npm.getMetricData(&resultSummary, udpLatencyAverage, latency)
+		resultSummary.protocol = protocolUDP
+	case protocolHTTP:
+		npm.getMetricData(&resultSummary, httpResponseTime, responseTime)
+		resultSummary.protocol = protocolHTTP
+	}
+	resultSummary.service = "P2P"
+	resultSummary.podRatio = npm.podRatioType
+	npm.resultSummary = resultSummary
+}
+
+func (npm *networkPerformanceMeasurement) getMetricData(data *testResultSummary, metricIndex int, metricName string) {
+	var metricDataItem measurementutil.DataItem
+	metricDataItem.Data = make(map[string]float64)
+	metricDataItem.Labels = make(map[string]string)
+	metricDataItem.Labels["Metric"] = metricName
+	npm.calculateMetricDataValue(&metricDataItem, metricIndex)
+	metricDataItem.Unit = metricUnitMap[metricDataItem.Labels["Metric"]]
+	data.DataItems = append(data.DataItems, metricDataItem)
+	klog.V(3).Infof("TestResultSummary: %v", data)
+}
+
+func (wp *workerPodInfo) formUniquePodPair() {
```

Comment:
Please create a dedicated `worker_pod_info.go` file and move
all workerPodInfo type definition together with all its methods (and node heap) to this file

[you can even move that to a separate PR - that would make reviewing easier]

Reply:
modified